src/fsharp/FSharp.Core/prim-types.fsi

```diff
@@ -1930,6 +1930,12 @@ namespace Microsoft.FSharp.Collections
         /// <param name="endIndex">The end index.</param>
         /// <returns>The sub list specified by the input indices.</returns>
         member GetSlice : startIndex:int option * endIndex:int option -> 'T list  
+
+        /// <summary>Get the index corresponding to ^i.</summary>
+        /// <param name="dim">The dimension of the index.</param>
```

Comment:
Instead of `dim`/`dimension`, we should use the term `rank` for everything. This goes for all other comments, parameter names in the PR.